### PR TITLE
RHAIENG-5858: feat(manifests) Keep 2025.2 runtime images for 3.5 GA

### PR DESCRIPTION
**Pull request by atheo89**
Related to: https://redhat.atlassian.net/browse/RHAIENG-5878 

## Description

Add new ImageStream manifests under manifests/rhoai/base/ to support the 2025.2 workbenches and their corresponding runtime images, ensuring that pipeline runtime images remain available and selectable in JupyterLab.

## How Has This Been Tested?

 Manifest validation success check:  `kustomize build manifests/rhoai/base > rhoai-manifests.yaml` executed without errors, placeholer on imagestreams assigned the proper image hashes 
 
 On cluster:
 
Screenshot 2026-06-29 at 11 56 00

`pipeline-runtime-images` ConfigMap generated by the notebook-controller

```
kind: ConfigMap
apiVersion: v1
metadata:
  name: pipeline-runtime-images
  namespace: atheo
  labels:
    opendatahub.io/managed-by: workbenches
  managedFields:{}
data:
  runtime-pytorch-rocm-python-3.12.json: '{"display_name":"Runtime | Pytorch | ROCm | Python 3.12","metadata":{"display_name":"Runtime | Pytorch | ROCm | Python 3.12","image_name":"registry.redhat.io/rhoai/odh-pipeline-runtime-pytorch-rocm-py312-rhel9@sha256:9fdc288e50e3c3dd794fe68d765add6fbf619fa8032599fc969c49273956d3f6","pull_policy":"IfNotPresent","tags":["rocm-pytorch"]},"schema_name":"runtime-image"}'
  runtime-pytorch-cuda-python-3.12-2025.2.json: '{"display_name":"Runtime | Pytorch | CUDA | Python 3.12 | 2025.2","metadata":{"display_name":"Runtime | Pytorch | CUDA | Python 3.12 | 2025.2","image_name":"registry.redhat.io/rhoai/odh-pipeline-runtime-pytorch-cuda-py312-rhel9@sha256:8b059881d8d7ef85883c7672a4464368ae564f2ce6104d54ebb03ac5d152d536","pull_policy":"IfNotPresent","tags":["pytorch-2025.2"]},"schema_name":"runtime-image"}'
  runtime-datascience-cpu-python-3.12-2025.2.json: '{"display_name":"Runtime | Datascience | CPU | Python 3.12 | 2025.2","metadata":{"display_name":"Runtime | Datascience | CPU | Python 3.12 | 2025.2","image_name":"registry.redhat.io/rhoai/odh-pipeline-runtime-datascience-cpu-py312-rhel9@sha256:22ea9fc4352ec427a240fb983afb0ad3e7c1b2f2dc97bce4587695777f23ef07","pull_policy":"IfNotPresent","tags":["datascience-2025.2"]},"schema_name":"runtime-image"}'
  runtime-tensorflow-cuda-python-3.12-2025.2.json: '{"display_name":"Runtime | TensorFlow | CUDA | Python 3.12 | 2025.2","metadata":{"display_name":"Runtime | TensorFlow | CUDA | Python 3.12 | 2025.2","image_name":"registry.redhat.io/rhoai/odh-pipeline-runtime-tensorflow-cuda-py312-rhel9@sha256:917272fc0483b20ac426626f5ee5bcdcbc4c88b5bbdbf00cf00a7a59d6bf5686","pull_policy":"IfNotPresent","tags":["tensorflow-2025.2"]},"schema_name":"runtime-image"}'
  runtime-minimal-cpu-python-3.12-2025.2.json: '{"display_name":"Runtime | Minimal | CPU | Python 3.12 | 2025.2","metadata":{"display_name":"Runtime | Minimal | CPU | Python 3.12 | 2025.2","image_name":"registry.redhat.io/rhoai/odh-pipeline-runtime-minimal-cpu-py312-rhel9@sha256:d3d4d82d0685feb25ec91cf8928dc5c5859436f44a21cd93b5d2ce7c1869a603","pull_policy":"IfNotPresent","tags":["minimal-2025.2"]},"schema_name":"runtime-image"}'
  runtime-pytorch-llm-compressor-cuda-python-3.12-2025.2.json: '{"display_name":"Runtime | PyTorch LLM Compressor | CUDA | Python 3.12 | 2025.2","metadata":{"display_name":"Runtime | PyTorch LLM Compressor | CUDA | Python 3.12 | 2025.2","image_name":"registry.redhat.io/rhoai/odh-pipeline-runtime-pytorch-llmcompressor-cuda-py312-rhel9@sha256:7c2ce6c9e93266a11cbfdef8b3abbeb9ffd13f983b9fb3a091607d1065d597e0","pull_policy":"IfNotPresent","tags":["pytorch-2025.2"]},"schema_name":"runtime-image"}'
  runtime-pytorch-llm-compressor-cuda-python-3.12.json: '{"display_name":"Runtime | PyTorch LLM Compressor | CUDA | Python 3.12","metadata":{"display_name":"Runtime | PyTorch LLM Compressor | CUDA | Python 3.12","image_name":"registry.redhat.io/rhoai/odh-pipeline-runtime-pytorch-llmcompressor-cuda-py312-rhel9@sha256:bb82e65682c3e36fe1397fb30272ebc56b2f23301015e4ac8977a814ba72e9b9","pull_policy":"IfNotPresent","tags":["pytorch"]},"schema_name":"runtime-image"}'
  runtime-minimal-cpu-python-3.12.json: '{"display_name":"Runtime | Minimal | CPU | Python 3.12","metadata":{"display_name":"Runtime | Minimal | CPU | Python 3.12","image_name":"registry.redhat.io/rhoai/odh-pipeline-runtime-minimal-cpu-py312-rhel9@sha256:eac5691f55bf797fc5992a1b45b4a21e743296d5d37a234eb112dca30e2e1b7c","pull_policy":"IfNotPresent","tags":["minimal"]},"schema_name":"runtime-image"}'
  runtime-tensorflow-cuda-python-3.12.json: '{"display_name":"Runtime | TensorFlow | CUDA | Python 3.12","metadata":{"display_name":"Runtime | TensorFlow | CUDA | Python 3.12","image_name":"registry.redhat.io/rhoai/odh-pipeline-runtime-tensorflow-cuda-py312-rhel9@sha256:e8c44fcdc152a69c38014106428a5f1de8c70dad0ec8c06f3a2f782dfa9d25fa","pull_policy":"IfNotPresent","tags":["tensorflow"]},"schema_name":"runtime-image"}'
  runtime-datascience-cpu-python-3.12.json: '{"display_name":"Runtime | Datascience | CPU | Python 3.12","metadata":{"display_name":"Runtime | Datascience | CPU | Python 3.12","image_name":"registry.redhat.io/rhoai/odh-pipeline-runtime-datascience-cpu-py312-rhel9@sha256:b0b8427da8303c46f0879b6064ded1aee2de143c756f9211ceb631f05a1c4919","pull_policy":"IfNotPresent","tags":["datascience"]},"schema_name":"runtime-image"}'
  runtime-pytorch-cuda-python-3.12.json: '{"display_name":"Runtime | Pytorch | CUDA | Python 3.12","metadata":{"display_name":"Runtime | Pytorch | CUDA | Python 3.12","image_name":"registry.redhat.io/rhoai/odh-pipeline-runtime-pytorch-cuda-py312-rhel9@sha256:ff10b6182910bdcf5018358ff884c4960c76d2536472da9966f797a98dbe6a9a","pull_policy":"IfNotPresent","tags":["pytorch"]},"schema_name":"runtime-image"}'
  runtime-pytorch-rocm-python-3.12-2025.2.json: '{"display_name":"Runtime | Pytorch | ROCm | Python 3.12 | 2025.2","metadata":{"display_name":"Runtime | Pytorch | ROCm | Python 3.12 | 2025.2","image_name":"registry.redhat.io/rhoai/odh-pipeline-runtime-pytorch-rocm-py312-rhel9@sha256:d9c32296f9b8b6b9b0fccaafe0e910d96e3e700fa14876f19ad7570523b4b384","pull_policy":"IfNotPresent","tags":["rocm-pytorch-2025.2"]},"schema_name":"runtime-image"}'
  runtime-tensorflow-rocm-python-3.12-2025.2.json: '{"display_name":"Runtime | TensorFlow | ROCm | Python 3.12 | 2025.2","metadata":{"display_name":"Runtime | TensorFlow | ROCm | Python 3.12 | 2025.2","image_name":"registry.redhat.io/rhoai/odh-pipeline-runtime-tensorflow-rocm-py312-rhel9@sha256:a4cfdf3f66e855efd3809f83e806becd5f30a8e18e73ef157f81aab37f07cd47","pull_policy":"IfNotPresent","tags":["rocm-tensorflow-2025.2"]},"schema_name":"runtime-image"}'
  runtime-tensorflow-rocm-python-3.12.json: '{"display_name":"Runtime | TensorFlow | ROCm | Python 3.12","metadata":{"display_name":"Runtime | TensorFlow | ROCm | Python 3.12","image_name":"registry.redhat.io/rhoai/odh-pipeline-runtime-tensorflow-rocm-py312-rhel9@sha256:d3c7ea1c6c8c3c8c29525a7146e066d49722d155dcf719f12e75cc7c041b6207","pull_policy":"IfNotPresent","tags":["rocm-tensorflow"]},"schema_name":"runtime-image"}'

```
 

Self checklist (all need to be checked):
- [x] Ensure that you have run `make test` (`gmake` on macOS) before asking for review

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

## Summary by CodeRabbit

* **New Features**
  * Added support for the 2025.2 runtime image set across multiple environments, including Data Science, Minimal, PyTorch, LLM Compressor, ROCm PyTorch, ROCm TensorFlow, and TensorFlow.
  * Updated runtime image references so the latest prebuilt images can be selected automatically.
  * Expanded runtime configuration handling to support versioned runtime variants consistently.

### CODE REVIEW

**Comment by atheo89** [comment]
@coderabitai full review

File: manifests/tools/generate_kustomization.py [hunk 1]
```diff
@@ -336,9 +337,9 @@ def _workbench_commit_replacements(wb: Workbench) -> list[str]:
 
 
 def _runtime_params_replacement(rt: Runtime) -> str:
-    """Single image-params replacement for a runtime (N only)."""
+    """Single image-params replacement for a runtime."""
     return _replacement_block(
-        f"{rt.param_key}-n",
+        f"{rt.base_key}{rt.suffix}",
```
[the remark this answers is not in the document]
Comment (jiridanek) [reply manifests/tools/generate_kustomization.py:342]: @atheo89 lets investigate this, @coderabbitai show the relevant code and do a detailed walkthrough for me

**Review by jiridanek — APPROVED** [review]
/lgtm, but let's investigate what coderabbitai says about the operator... I believe it's wrong, because the operator deals with params-latest.env and this is all in params.env

File: manifests/tools/generate_kustomization.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jiridanek) [reply manifests/tools/generate_kustomization.py:342]: @atheo89 rabbit realized its error

File: manifests/rhoai/base/runtime-rocm-pytorch-imagestream-2025-2.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+---
+apiVersion: image.openshift.io/v1
+kind: ImageStream
+metadata:
+  labels:
+    opendatahub.io/runtime-image: "true"
+  annotations:
+    opendatahub.io/runtime-image-url: "https://github.com/opendatahub-io/notebooks/tree/main/runtimes"
+    opendatahub.io/runtime-image-name: "Runtime | Pytorch | ROCm | Python 3.12 | 2025.2"
```
[the remark this answers is not in the document]
Comment (jiridanek) [reply manifests/rhoai/base/runtime-rocm-pytorch-imagestream-2025-2.yaml:9]: @atheo89 wdyt?

File: manifests/tools/generate_kustomization.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (atheo89) [reply manifests/tools/generate_kustomization.py:342]: The operator map in workbenches.go only patches params-latest.env keys ending in -n, replacing dummy values with RELATED_IMAGE_* from the CSV. (applies also for this [review ](https://github.com/opendatahub-io/notebooks/pull/3971#discussion_r3492107662)) That’s the path for current/latest shipping images.

The 2025.2 runtime keys are in params.env with pinned catalog digests (v2.25), same pattern as the 2025.2 workbench entries already on this branch. They are not meant to go through ApplyParams().

Selectability comes from the ImageStream manifests themselves: separate runtime-*-2025-2 ImageStreams, kustomize replacements from notebook-image-params, and the opendatahub.io/runtime-image label/annotations and from the `odh-notebook-controller`. 

So there is a cross-repo gap — the odh-operator doesn’t know about -2025-2 runtime keys — but it only matters if we later want operator-driven auto-updates for versioned runtime images. That’s follow-up scope, not a blocker for keeping 2025.2 runtimes available in 3.5 GA.

**Comment by atheo89** [comment]
closing this in favor: https://github.com/opendatahub-io/notebooks/pull/3973
